pycbc/detector.py

```diff
@@ -322,3 +304,65 @@ def overhead_antenna_pattern(right_ascension, declination, polarization):
 def effective_distance(distance, inclination, f_plus, f_cross):
     return distance / np.sqrt( ( 1 + np.cos( inclination )**2 )**2 / 4 * f_plus**2 + np.cos( inclination )**2 * f_cross**2 )
 
+def dist_lisa_gbd(det, ref_time):
+    L_pos = LISA().get_pos(ref_time)
+    if isinstance(det, str):
+        det = Detector(det, ref_time).location
+        det = (det*units.m / units.AU).decompose()
+    if isinstance(det, np.ndarray) and det.shape[0] == 3:
+        det = det
+    dist = np.array([det[0] - L_pos[0],
+                     det[1] - L_pos[1],
+                     det[2] - L_pos[2]])
+    return dist
+
+
+"""     LISA class      """
+
+
+class LISA(object):
+    def __init__(self):
+        None
+
+    def get_pos(self, t_gps):
+        t_gps = Time(val=t_gps, format='gps', scale='utc').jyear
+
+        n = np.array(range(1, 4))
+        kappa, _lambda_ = 0, 0
+        alpha = 2. * np.pi * t_gps/1 + kappa
+        beta_n = (n - 1) + (2. * np.pi/3) + _lambda_
+        a, L = 1., .1
+        e = L/(2. * a * np.sqrt(3))
+        prod = a*e*(sin(alpha)*cos(alpha)*sin(beta_n))
+
+        x = a*cos(alpha) + prod - (1 + sin(alpha)**2)*cos(beta_n)
+        y = a*sin(alpha) + prod - (1 + cos(alpha)**2)*sin(beta_n)
+        z = -np.sqrt(3)*a*e*cos(alpha - beta_n)
+        self.location = np.array([x, y, z])
+
+        return self.location
+
+    def time_delay_from_location(self, other_location, right_ascension,
+                                 declination, t_gps):
+        dx = dist_lisa_gbd(other_location, t_gps)
```

Comment:
Instead of creating a function that exists on its own and needs switch statements, a more class based approached would be to have both the LISA class and the Detector class have a .get_icrs_pos method. Then you just call both those methods and use the output to calculate the distance between them.